### PR TITLE
Fix return type for DataFrame.groupBy / cube / rollup

### PR DESCRIPTION
**Pull request by dangercrow**
2.3 has these data types and I was erroneously gettting errors for them.

Note this is a port of e2d225f06ff36fcbf79e2123f1c18f380e862728

I tried a cherry-pick but it had some issues (not sure why)

### CODE REVIEW

File: third_party/3/pyspark/sql/dataframe.pyi [hunk 1]
```diff
@@ -91,17 +92,17 @@ class DataFrame:
     def selectExpr(self, *expr: List[str]) -> 'DataFrame': ...
     def filter(self, condition: ColumnOrName) -> 'DataFrame': ...
     @overload
-    def groupBy(self, *cols: ColumnOrName) -> 'DataFrame': ...
+    def groupBy(self, *cols: ColumnOrName) -> GroupedData': ...
```
Comment (zero323) [inline third_party/3/pyspark/sql/dataframe.pyi]: There is a dangling `'` at the end of the line here.

**Comment by zero323** [comment]
Thanks for the PR @dangercrow 

> I tried a cherry-pick but it had some issues (not sure why)

There is quite a bit of history between these two, and cherry picking might get tricky.

Please make sure that your PR passes CI build.

File: third_party/3/pyspark/sql/dataframe.pyi (hunk 1 above)
Comment (zero323) [inline third_party/3/pyspark/sql/dataframe.pyi]: There is a dangling quote in the return type annotation here.

Comment (dangercrow) [reply third_party/3/pyspark/sql/dataframe.pyi]: Yeah, I did it in a text editor as I didn't have a proper IDE. Thanks for taking a look, it was on my backlog!

**Comment by dangercrow** [comment]
FYI I did also have plans to update the Column.__eq__ signature as I think I ran into some issues with that also.

**Comment by zero323** [comment]
> FYI I did also have plans to update the `Column.__eq__` signature as I think I ran into some issues with that also.

Please be sure to check https://github.com/zero323/pyspark-stubs/issues/2 first. At the moment all annotations for comparison dunder methods are explicitly ignored.

**Comment by zero323** [comment]
LGTM, thank you. Merged into branch-2.3,

**Pull request merged**